AndroidManifest.xml

```diff
@@ -21,6 +21,8 @@
   <uses-permission android:name="android.permission.CAMERA"/>
 
   <uses-feature android:name="android.hardware.camera" />
+  <uses-feature android:name="android.hardware.location.gps" />
+  <uses-feature android:name="android.hardware.location.network" />
```

Comment:
The app specifies in the manifest what will be using to have access to specific features, correct? Does that mean that until now those two were not used?

Reply:
It's an example of Google's changing APIs and specs - basically since sdk 21, when you want to use location services, you need to explicitly declare in the manifest that the app will be using this or that hardware feature. Previously it wasn't required, and the reasoning for the change is that they want to filter out devices that simply don't have those features. Because probably there were scenarios, when someone could install an app that uses a camera on a device without one, for example.